src/puppetlabs/puppetdb/utils.clj

```diff
@@ -257,3 +257,10 @@
       {:href (to-href data)}
       (-> (sutils/parse-db-json data)
           (update :href to-href)))))
+
+(defn hsql?
+  "given the globals configuration bundle, determine if hsqldb is being used."
+  [globals]
```

Comment:
I think I must be misunderstanding what you're proposing here, but just to be clear, this is a different map than the db-spec map I described above -- it has different keys, different values, and db-spec is not contained anywhere inside it. Globals really seems more appropriate than db or db-spec to me. When we use db, or *db* in other functions, we're talking about that db-spec object.